src/interpreter/plugin/FunctionPlugin.ts

```diff
@@ -98,10 +114,16 @@ export enum ArgumentTypes {
 export interface FunctionArgument {
   argumentType: ArgumentTypes,
 
+  /**
+   * Argument should be passed with full type information.
+   * (e.g. Date/DateTime/Time/Currency/Percentage for numbers)
+   */
+  passSubtype?: boolean,
```

Comment:
> If we know that the argument has a returnNumberType set, we may assume that it should be passed as a full type?

Not really. Most functions ignore input types. Some functions (usualy logical ones) use the type, e.g. IF etc.
Some other function set output type (e.g. financial).

Reply:
So shouldn't this be up to the function to decide?
I believe it already does but based on some metadata to the Core, which may be confusing and adds to the complexity of custom functions

Reply:
> So shouldn't this be up to the function to decide?

Why not keep it in metadata, since it is a pretty automatic decision (you either want raw data, or rich data).

Reply:
I don't have better arguments. Metadata is fine.